### PR TITLE
Remove Foreman 3.3 and 3.4 deprecation warnings

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -201,8 +201,7 @@ def self.default_settings
 
   def self.load_defaults
     return false unless table_exists?
-    Foreman::Deprecation.deprecation_warning('3.4', "subclassing Setting is deprecated '#{name}' should be migrated to setting DSL "\
```
Comment (ekohl) [inline app/models/setting.rb:204]: This only removes the deprecation. Perhaps @ezr-ondrej can enlighten us, but I think the proper fix is to remove the whole method.

File: app/services/proxy_api/resource.rb [hunk 2]
```diff
@@ -67,7 +67,6 @@ def parse(response)
     # Perform GET operation on the supplied path
     def get(path = nil, payload = {})
       query = payload.delete(:query)
-      Foreman::Deprecation.deprecation_warning("3.3", "passing additional headers to ProxyApi resource GET action") unless payload.empty?
```
Comment (ekohl) [inline app/services/proxy_api/resource.rb:70]: I think you should remove the code that deals with a non-empty payload.

File: app/services/setting_registry.rb [hunk 3]
```diff
@@ -159,7 +159,6 @@ def load_values(ignore_cache: false)
 
   def _add(name, category:, type:, default:, description:, full_name:, context:, value: nil, encrypted: false, collection: nil, options: {})
     select_collection_registry.add(name, collection: collection, **options) if collection
-    Foreman::Deprecation.deprecation_warning('3.3', "initial value of setting '#{name}' should be created in a migration") if value
```
Comment (ekohl) [inline app/services/setting_registry.rb:162]: Here it means that the code should no longer handle a `value`. It should remove the method parameter `value` and anything that deals with it.

There is also an entry that masks this:
https://github.com/theforeman/foreman/blob/d4569bb4de9b6f2cc4972452e4e19b027c522bcf/config/as_deprecation_whitelist.yaml#L3
You should first make sure that's gone.

**Review by ekohl — CHANGES_REQUESTED** [review]
Removing deprecations doesn't mean blindly removing the warning. It means removing the compatibility code.

File: app/services/proxy_api/resource.rb (hunk 2 above)
Comment (upadhyeammit) [reply app/services/proxy_api/resource.rb:70]: hmm, then to get rid of the payload for get method it looks some of the classes inherited from `ProxyAPI::Resource` needs change. I will git it a try

**Comment by ekohl** [comment]
Closing this now, but it still needs to be done properly.

**Pull request closed without merging**